src/espnow.rs

```diff
@@ -0,0 +1,183 @@
+use log::info;
+
+use esp_idf_hal::mutex::Mutex;
+use esp_idf_sys::*;
+
+static RECV_CALLBACK: Mutex<Option<Box<dyn FnMut(&[u8], &[u8]) + Send>>> = Mutex::new(None);
```

Comment:
I think it would be better if this state is part of the `EspNowClient` structure. For one, that would allow you to get rid of the mutexes in case you decide to supply the callbacks during construction time, which I think is the simplest and cleanest approach.

Reply:
yeah, making callbacks storage a part of `EspNowClient` struct is something I was aiming for originally but seems it's not possible in `rust` since I do not have a way to pass `context` (self) with the function signature (https://github.com/esp-rs/esp-idf-svc/pull/58/files#diff-fab8c30d89ae294965502f35d1d766710c2a9970398b2be902f0872948300ee9R154) that `esp_now_register_recv_cb` expects. AFAICT there's no user_ptr (`*mut c_void`) that I could leverage in order to avoid the global state variable. Let me know if you think there's a technique I could use to workaround this - I was not able to come up with a solution :(